src/components/Common/SensitiveInput/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Field } from 'formik';
+import React, { useState } from 'react';
+import EyeOffIcon from '../../../assets/eye-off.svg';
+import EyeIcon from '../../../assets/eye.svg';
+
+interface CustomInputProps extends React.ComponentProps<'input'> {
+  as?: 'input';
+  isLastButton?: boolean;
+}
+
+interface CustomFieldProps extends React.ComponentProps<typeof Field> {
+  as?: 'field';
+  isLastButton?: boolean;
+}
+
+type SensitiveInputProps = CustomInputProps | CustomFieldProps;
+
+const SensitiveInput = ({
+  isLastButton = true,
+  as = 'input',
+  ...props
+}: SensitiveInputProps): JSX.Element => {
```

Comment:
It's better to type this as a `React.FC` to get the correct typing for children and possible return types. (Even though we know the return type here, it's best to keep this component consistent with how we do all our other components)

Reply:
As we don't allow passing children here it would make more sense to not type it as if children were allowed?

Reply:
Children are always allowed, even if they don't do anything.